core/io/resource_format_binary.cpp

```diff
@@ -825,10 +823,7 @@ Error ResourceLoaderBinary::load() {
 
 			Variant value;
 
-			error = parse_variant(value);
-			if (error) {
-				return error;
-			}
+			RETURN_IF_ERR(parse_variant(value));
```

Comment:
Incorrect as well, this mutates a class variable not a local one, should be:
```suggestion
			error = parse_variant(value);
			RETURN_IF_ERR(error);
```